### DISCUSSION TITLE
When configuring disk_free_limit.absolute using KB, MB, or GB as units for the limit.absolute parameter, the final result after starting rabbitmq is incorrect

### DISCUSSION THREAD

**Discussion by charlienss** [rabbitmq/rabbitmq-server#9780] [Other] [answered]
### Describe the bug

When configuring disk_free_limit.absolute using KB, MB, or GB as units for the limit.absolute parameter, the final result after starting rabbitmq is incorrect

### Reproduction steps

RabbitMQ Version：3.12.2
1.  configuring disk_free_limit.absolute as 50MB
2.  the final result is 48MB after starting rabbitmq

### Expected behavior

I hope that when configuring disk_free_limit.absolute parameter is in KB, MB, or GB units, the result is correct after starting rabbitmq

**Comment by michaelklishin** [comment, marked as answer]
You do not specify how exactly you verify what the configured value is. 

We do not guess in this community but this one time I will: the discrepancy very likely comes down to the difference between the units and rounding behaviors used by various tools: [mebibytes vs. megabytes, for example](https://en.wikipedia.org/wiki/Megabyte).

RabbitMQ does not do [anything particularly creative](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit_common/src/rabbit_resource_monitor_misc.erl#L17-L34) when it comes to computing the value in bytes.

This difference won't matter in practice, and in particular it won't matter when [more reasonable free disk space watermark values](https://rabbitmq.com/production-checklist.html#monitoring-and-resource-usage) are used (such as 1 GiB or 1 GB).  50 MiB is about as good as not having any watermark, in particular with modern features like quorum queues and streams.

The reason why it is still the default comes down to this: with a lot of installations on developer machines, the volume dedicated to RabbitMQ's data directory can be very small, so a low watermark of 1 GiB can instantly trigger an alarm and block all publishers. Which does not exactly help when
you try running a tutorial.

**Comment by michaelklishin** [comment]
I suspect that what you use to verify the value is the management UI. It uses MiB and rounds up to the nearest MiB, so

50000000 bytes in MiB with rounding is 48 MiB, what the management UI displays.

Setting the value to 1 GiB makes the UI display 1024 MiB. Setting it to 1 GB will display 954 MiB (953.6 rounded up).

So there is nothing to fix, and it makes no practical difference.

**Reply by charlienss** [reply]
Suggest writing a note on the official website as an example to prevent errors caused by the use of MB and MIB units.
https://www.rabbitmq.com/configure.html#configuration-files
![image](https://github.com/rabbitmq/rabbitmq-server/assets/35131874/853ae69b-9e63-432a-ac35-48f6d1e6c0c3)
